test/src/main/java/org/apache/accumulo/test/functional/WALSunnyDayIT.java

```diff
@@ -258,17 +258,27 @@ private int getWaitFactor() {
     return waitLonger;
   }
 
-  private Map<String,Boolean> _getWals(Connector c) throws Exception {
-    Map<String,Boolean> result = new HashMap<>();
-    Instance i = c.getInstance();
-    ZooReaderWriter zk = new ZooReaderWriter(i.getZooKeepers(), i.getZooKeepersSessionTimeOut(),
-        "");
-    WalStateManager wals = new WalStateManager(c.getInstance(), zk);
-    for (Entry<Path,WalState> entry : wals.getAllState().entrySet()) {
-      // WALs are in use if they are not unreferenced
-      result.put(entry.getKey().toString(), entry.getValue() != WalState.UNREFERENCED);
+  static Map<String,WalState> _getWals(Connector c) throws Exception {
+    while (true) {
+      try {
+        Map<String,WalState> result = new HashMap<>();
+        Instance i = c.getInstance();
+        ZooReaderWriter zk = new ZooReaderWriter(i.getZooKeepers(), i.getZooKeepersSessionTimeOut(),
+            "");
+        WalStateManager wals = new WalStateManager(c.getInstance(), zk);
+        for (Entry<Path,WalState> entry : wals.getAllState().entrySet()) {
+          // WALs are in use if they are not unreferenced
+          result.put(entry.getKey().toString(), entry.getValue());
+        }
+        return result;
+      } catch (WalMarkerException wme) {
+        if (wme.getCause() instanceof NoNodeException) {
```

Comment:
TODO this should probably be handled in WalStateManager.. need to open an issue.

Reply:
I determined this should probably not be handled in WalStateManager.  However while looking into the issue I discovered #866